analysis_methods.py

```diff
@@ -0,0 +1,78 @@
+import random
+import math
+from ms2deepscore import MS2DeepScore
+
+def create_sets(master_subset, size_training, size_validation, size_testing):
+    """this method randomly splits a subset into user-defined portions for training, validation and testing.
+    the arguments for size_training, size_validation and size_testing are expected to be percentages e.g. 0.8. Returns
+    a trainingset, validationset, testingset"""
+    testing_amount = math.floor(size_testing*len(master_subset))
+    validation_amount = math.floor(size_validation*len(master_subset))
+    training_amount = math.floor(size_training*len(master_subset))
+    testing_set = []
+    validation_set = []
+    training_set = []
+    currentset = master_subset.copy()
+    while len(currentset) != 0:
+        test_proposed = random.sample(currentset, testing_amount)
+        for spectrum in test_proposed:
+            testing_set.append(spectrum)
+            currentset.remove(spectrum)
+        validation_proposed = random.sample(currentset, validation_amount)
+        for s in validation_proposed:
+            validation_set.append(s)
+            currentset.remove(s)
+        training_proposed = random.sample(currentset, training_amount)
+        for i in training_proposed:
+            training_set.append(i)
+            currentset.remove(i)
+    return training_set, validation_set, testing_set
+
+def create_pairs(testingset):
+    """takes a dataset as its argument and returns a list of randomly paired spectra. If len(testingset) not divisible

```

Comment:
What is the goal of this function? Is it for calculating an RMSE for the test set?
It creates pairs of spectra at random, but only chooses each spectrum for 1 pair. If the goal is to calculate the RMSE a good alternative is just comparing everything to everything. So for 1000 spectra this would result in 1000.000 combinations.

Reply:
I just saw the use in the next function, so I think it is best to change this to comparing everything to everything. Luckily MS2Deepscore is really fast, so this is doable

Reply:
See comments on next functions as well